### PR TITLE
Fix the document and return checks of several APIs

### PR DESCRIPTION
**Pull request by PeiweiHu**
##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man3/BIO_ctrl.pod [hunk 1]
```diff
@@ -82,9 +82,9 @@ return a size_t type and are functions, BIO_pending() and BIO_wpending() are
 macros which call BIO_ctrl().
 
 BIO_get_ktls_send() returns 1 if the BIO is using the Kernel TLS data-path for
-sending. Otherwise, it returns zero.
+sending. Otherwise, it returns zero or a negative number.
```
Comment (slontis) [inline doc/man3/BIO_ctrl.pod:85]: This change is not correct... The macro's used are

````
#  define BIO_get_ktls_send(b)         \
     (BIO_ctrl(b, BIO_CTRL_GET_KTLS_SEND, 0, NULL) > 0)
#  define BIO_get_ktls_recv(b)         \
     (BIO_ctrl(b, BIO_CTRL_GET_KTLS_RECV, 0, NULL) > 0)
````

Comment (slontis) [reply doc/man3/BIO_ctrl.pod:85]: i.e. they return 0 or 1.

File: doc/man3/EVP_EncryptInit.pod [hunk 2]
```diff
@@ -1258,7 +1258,8 @@ EVP_CIPHER_CTX_is_encrypting() returns 1 if the I<ctx> is set up for encryption
 EVP_CIPHER_param_to_asn1() and EVP_CIPHER_asn1_to_param() return greater
 than zero for success and zero or a negative number on failure.
 
-EVP_CIPHER_CTX_rand_key() returns 1 for success.
+EVP_CIPHER_CTX_rand_key() returns 1 for success and zero or a negative number
```
Comment (slontis) [inline doc/man3/EVP_EncryptInit.pod:1261]: Can you prove that this returns negative..
The paths I can see that use this (including OSSL_CIPHER_PARAM_RANDOM_KEY) seem to return 0 or 1

File: doc/man3/EVP_VerifyInit.pod [hunk 3]
```diff
@@ -50,7 +50,7 @@ EVP_VerifyInit_ex() and EVP_VerifyUpdate() return 1 for success and 0 for
 failure.
 
 EVP_VerifyFinal_ex() and EVP_VerifyFinal() return 1 for a correct
-signature, 0 for failure and -1 if some other error occurred.
+signature, 0 for failure and a negative number if some other error occurred.
```
Comment (slontis) [inline doc/man3/EVP_VerifyInit.pod:53]: Agree with this change since it can return -2

File: doc/man3/EVP_EncryptInit.pod (hunk 2 above)
Comment (PeiweiHu) [reply doc/man3/EVP_EncryptInit.pod:1261]: Thanks for your review. `EVP_CIPHER_CTX_rand_key` can invoke and return `EVP_CIPHER_CTX_ctrl`, which contains negative return values. 

Why  `EVP_CIPHER_CTX_ctrl` contains negative values:
+ every check of it is in the format like  `EVP_CIPHER_CTX_ctrl <= 0`
+ it contains the code snippet like:
```
   if (ret <= 0)
        goto end;
 
   if (arg < 0)
        return -1;

   if (set_params)
       ret = evp_do_ciph_ctx_setparams(ctx->cipher, ctx->algctx, params);
   else
       ret = evp_do_ciph_ctx_getparams(ctx->cipher, ctx->algctx, params);
   goto end;
```
According to `crypto/evp/evp_local.h`, `evp_do_ciph_ctx_setparams` has the following return values:
```
/*
 * These methods implement different ways to pass a params array to the
 * provider.  They will return one of these values:
 *
 * -2 if the method doesn't come from a provider
 *    (evp_do_param will return this to the called)
 * -1 if the provider doesn't offer the desired function
 *    (evp_do_param will raise an error and return 0)
 * or the return value from the desired function
 *    (evp_do_param will return it to the caller)
 */
```

File: doc/man3/BIO_ctrl.pod (hunk 1 above)
Comment (PeiweiHu) [reply doc/man3/BIO_ctrl.pod:85]: Thanks for your review. `BIO_ctrl` can return -1 and -2. `BIO_get_ktls_send/recv` can also return these values.

```
long BIO_ctrl(BIO *b, int cmd, long larg, void *parg)
{
    long ret;

    if (b == NULL)
        return -1;
    if (b->method == NULL || b->method->ctrl == NULL) {
        ERR_raise(ERR_LIB_BIO, BIO_R_UNSUPPORTED_METHOD);
        return -2;
    }

    if (HAS_CALLBACK(b)) {
        ret = bio_call_callback(b, BIO_CB_CTRL, parg, 0, cmd, larg, 1L, NULL);
        if (ret <= 0)
            return ret;
    }

    ret = b->method->ctrl(b, cmd, larg, parg);

    if (HAS_CALLBACK(b))
        ret = bio_call_callback(b, BIO_CB_CTRL | BIO_CB_RETURN, parg, 0, cmd,
                                larg, ret, NULL);

    return ret;
}
```

Comment (PeiweiHu) [reply doc/man3/BIO_ctrl.pod:85]: Yes, you are right. They are updated this year. I found this last year and thought they still exist...... Thanks!

File: doc/man3/EVP_EncryptInit.pod (hunk 2 above)
Comment (slontis) [reply doc/man3/EVP_EncryptInit.pod:1261]: The code does do this....
 ```` 
 end:
    if (ret == EVP_CTRL_RET_UNSUPPORTED) {
        ERR_raise(ERR_LIB_EVP, EVP_R_CTRL_OPERATION_NOT_IMPLEMENTED);
        return 0;
    }
````

So it is picking up one of the cases..
The decision is then if it should return 0 in the case of -2..
I will let another reviewer have an opinion about this..

**Pull request closed without merging**